aten/src/ATen/core/type.cpp

```diff
@@ -1141,16 +1144,16 @@ std::string UnionType::unionStr(TypePrinter printer, bool is_annotation_str) con
       ss << NumberType::get()->str();
     }
   }
-  ss << "]";
+  ss << close_delimeter;
   return ss.str();
 }
 
 std::string UnionType::str() const {
-  return this->unionStr(nullptr, /*is_annotation_str=*/false);
+  return this->unionStr(nullptr, /*is_annotation_str=*/false, "(", ")");
```

Comment:
You can probably use `is_annotation_str` to branch on which delimiter to use.

Reply:
sounds good